src/runtime/lpython_builtin.py

```diff
@@ -198,18 +198,19 @@ def bin(n: i32) -> str:
         return '0b0'
     prep: str
     prep = '0b'
-    if n < 0:
-        n = -n
+    n_: i32 = n
+    if n_ < 0:
+        n_ = -n_
         prep = '-0b'
     res: str
     res = ''
-    if (n - _lpython_floordiv(n, 2)*2) == 0:
+    if (n_ - _lpython_floordiv(n_, 2)*2) == 0:
         res += '0'
     else:
         res += '1'
-    while n > 1:
-        n = _lpython_floordiv(n, 2)
-        if (n - _lpython_floordiv(n, 2)*2) == 0:
+    while n_ > 1:
+        n_ = _lpython_floordiv(n_, 2)
+        if (n_ - _lpython_floordiv(n_, 2)*2) == 0:
```

Comment:
It seems that the runtime library assigns to function parameters at several places. In this PR, I updated three such places. With respect to https://github.com/lcompilers/lpython/issues/1784#issuecomment-1546507259, I would like to know if we should update all places in the runtime library to not allow assignment to function parameters or if we should keep allowing assignment to function parameters.

Reply:
We should not allow assigning to function parameters, unless the parameter is marked Out or InOut, but that's only allowed for arrays. Integers in Python can only be In, and we should not allow assigning to them.

Reply:
Sure, got it.